src/client.js

```diff
@@ -169,46 +169,56 @@ assign(Client.prototype, {
     }
   },
 
-  poolDefaults(poolConfig) {
-    const name = this.dialect + ':' + this.driverName + ':' + this.__cid
+  poolDefaults() {
+    return {min: 2, max: 10, Promise}
+  },
+
+  getPoolSettings(poolConfig) {
+    poolConfig = defaults({}, poolConfig, this.poolDefaults());
+    // acquire connection timeout can be set on config or config.pool
+    // choose the smallest, positive timeout setting and set on poolConfig
+    const timeouts = [
+      this.config.acquireConnectionTimeout,
+      poolConfig.acquireTimeoutMillis
+    ].map(parseInt).filter(x => x > 0);
```

Comment:
Throwing an error on invalid parameters would be better than just filtering them out silently.

Reply:
Also there was default acquireConnectionTimeout of 60000, should it be added here?

Reply:
I agree that throwing would be better...

Previously, an invalid value for `config.acquireConnectionTimeout` would result in the timeout being `1` (default behavior when `delay` is not a valid, positive integer). If `pool.acquireTimeoutMillis` is invalid, generic-pool will throw `throw new Error('delay must be a positive int')` during acquisition.

I'd like to use joi, ajv or similar to define a schema for the config object to validate the options passed in the constructor, so that validation isn't sprinkled throughout the implementation. I could start with a very simple schema that covers just these properties, and others could extend the schema to include more validation later. How do you feel about that?

Alternatively, we can explicitly validate both settings here and throw an error if either one is ` value !== undefined && value !== null && (isNaN(parseInt(value)) || parseInt(value) < 0)`? I think that's the smallest possible variation from the current behavior that results in an error, though the error will be moved up to client construction from connection acquisition in the case that `pool.acquireTimeoutMillis` has an invalid value.

Reply:
Being able to validate config objects (and that way also document configuration better) with real validator sounds like a really good idea, but I think adding that deserves its own PR.

So I would go with alternative in this PR. Later on if proper config validation is added, many places in code can be revisited and checks that configuration vars are fine may be dropped.